File: src/dynamic/number_solitaire.py
def solution(A):
    # The first six items are used for padding only, so that we can have
    # a unified for loop, no matter how many squares are there in input.
    # The first item is never accessed.
    max_so_far = [A[0]] * (len(A) + 6)
    max_so_far
    l = len(max_so_far)
    l
    for index in range(1, len(A)):
        # Because we have a fixed length of window as 6, the time
        # complexity of max(max_so_far[index : index + 6]) is O(1).
        u = max_so_far[index + 6]
        u
        max_so_far[index + 6] = max(max_so_far[index : index + 6]) +  A[index]
        u = max_so_far[index + 6]
        u
    return max_so_far[-1]

arr = [2,-2, 0,9,-1,-2, 2, -3]
y = solution(arr)
y

File: src/dynamic/number_solitaire.js

function solution(arr) {
    let step = 0
    sum = arr[0]
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] > 1 ) {
            sum = sum + arr[i]
            step = i 
            step
        } else if(arr[i] < 1){
            checkNext6Numbers()

        } else if (i==arr.length-1) {
            sum = sum + arr[i]
        }
    }
    return sum
}

let arr = [1,-2, 0,9,-1,-2, 2, -3, -3, -5, -6, -7, -50000, -4,-6, -7, -8, -4, 8, -3, -5, 8, 8, 8]
let y = solution(arr)
y

// || i-step == 6